httpclient/metrics_round_tripper.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+package httpclient
+
+import (
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+var (
+	// ClassifyRequest does request classification, producing non-parameterized summary for given request.
+	ClassifyRequest func(r *http.Request, requestType string) string
```

Comment:
Please consider moving this package-level function to `MetricsRoundTripperOpts`. This would allow for greater flexibility, as we might want to use different classifiers for different HTTP clients. Additionally, with this change, passing `requestType` as an argument would no longer be necessary.